From 0000000000000000000000000000000000000000 Mon Sep 17 00:00:00 2001
From: Samuel Attard <samuel.r.attard@gmail.com>
Date: Thu, 28 Dec 2023 15:26:31 +1300
Subject: Always set --target for MacOS builds instead of -arch.

This allows MacOS targets to be built on Linux remotes via reclient
without any additional flag fiddling.

Bug: 1494896
Ref: https://chromium-review.googlesource.com/c/chromium/src/+/4966183

diff --git a/build/config/mac/BUILD.gn b/build/config/mac/BUILD.gn
index 85a668d33059766ac3ebc92b93039a2a7b383084..73f50eabdbfc47d845423f696e1fcdaf2e95af76 100644
--- a/build/config/mac/BUILD.gn
+++ b/build/config/mac/BUILD.gn
@@ -26,14 +26,7 @@ config("compiler") {
   } else {
     assert(false, "unknown current_cpu $current_cpu")
   }
-  if (host_os == "mac") {
-    common_mac_flags += [
-      "-arch",
-      clang_arch,
-    ]
-  } else {
-    common_mac_flags += [ "--target=$clang_arch-apple-macos" ]
-  }
+  common_mac_flags += [ "--target=$clang_arch-apple-macos" ]
 
   # This is here so that all files get recompiled after an Xcode update.
   # (defines are passed via the command line, and build system rebuild things
